### ISSUE TITLE
Hundreds of threads after day or more of activity

### ISSUE THREAD

**Issue by undersun81** [ikalchev/HAP-python#190]
Currently I have HAP server running for about 2 weeks and any request just fails because it can't create a new thread.

gdb attach <pid>
gdb info thread 
Shows 254 threads, mostly in recv for example
Thread 0x553c7470 (LWP 7985) "python3" 0x76f3e850 in recv () at ../sysdeps/unix/syscall-template.S:84

**Comment by undersun81** [comment]
I think I have a good guess about what is going on here.
Several Apple devices I have at home (iPhones, iPad and HomePod) periodically or by a user request ask HAP-server about state or subscribe. From logs I can see they use different client ports, but they(clients) never close these sockets and next time use another port and HAP-server creates new connection and new corresponding thread. Linux by default has very long timeout values for TCP-sockets, so eventually I have hundreds of threads and open connections.  So I hit the limit somewhere in python and it just fails  with error when trying to create new thread. This story happen to me every evening, then during night most of sockets timed out and I had proper usage during the day, and story repeated in the evening.

HAP uses blocking sockets, so my solution was to set timeout for a client socket when we connected.
So in hap_server.py in method HAPServer.get_request I added 
"client_socket.settimeout(60)" after "client_socket, client_addr = super(HAPServer, self).get_request()".

This change has fixed my problem, but I'm not an expert in HomeKit protocol and not sure about subscription case. May be clients which subscribed expect to hold connection as long as possible. If so then they should send some ping periodically into the same connection and not just start talking through another socket. Looks like apple devices just don't care to properly shutdown those sockets (may be in some specific case, like sleeping device) and they seem don't care if we have short timeout on them.

**Comment by niobos** [comment]
I don't think it has to do with open connections. I have TCP-keepalive configured, so dead connections get killed after about 5 minutes. Still, I see the number of threads increasing over time...
